datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -541,6 +542,8 @@ async fn fetch_statistics(
 pub struct ParquetSink {
     /// Config options for writing data
     config: FileSinkConfig,
+    /// File metadata from successfully produced parquet files.
```

Comment:
```suggestion
/// File metadata from successfully produced parquet files. The Mutex is only used to allow inserting to HashMap from behind borrowed reference in DataSink::write_all.
```

The use of a Mutex here is confusing without the context of this PR, so I think it would be a good idea to leave a comment explaining. 

I think this is a fine temporary workaround, but I'm sure we can find a way to return FileMetaData in a new public interface without breaking changes to DataSink or using locks.

Reply:
Absolutely agreed.